### ISSUE TITLE
Auto-delete merged PRs should be disabled by default

### ISSUE THREAD

**Issue by qbunt** [sindresorhus/refined-github#2710]
Auto deleting anything without the user's knowledge is a a pretty high-risk default, especially if the user cannot restore the branch. I did not know this was a feature of refined-github, and seeing this in action, I've since uninstalled.

I think it's a great feature for those who need it, so it should stay, but on first install it should definitely not be active.

**Comment by busches** [comment]
I've not seen an example of where you can delete the branch but not restore it, can you explain how this is possible?

**Comment by qbunt** [comment]
Some pull request pages don’t contain the restore button, so you have to go and either push again or something else just get an active branch back.

As I said, I’m sure it’s needed by some. Enabling this on without warning the user seems dangerous.


________________________________

**Comment by busches** [comment]
Do you have an example of where you deleted it, you being the extension,
and you can't restore it? This seems like a bug on GH.

**Comment by qbunt** [comment]
@busches, I don't have a specific example, our `develop` branch kept getting nuked and I couldn't figure out why.

**Comment by qbunt** [comment]
Looking at closed issues this has come up before and been closed with a similar discussion. Given that this keeps coming up, I think it's worth reconsidering as people keep getting surprised. I'll see myself out.

**Comment by busches** [comment]
That's why I'm after an example, I've never seen this behavior before and
worked on a project for a long time with a lot of long lived feature
branches.

You mentioned `develop` was getting nuked, is it a protected branch? And if
yes, are you still able to delete it? If that's the case, then we should
account for that and not auto delete a protected branch, just because
you're an admin.

**Comment by qbunt** [comment]
Many of the repos I'm on do not have branch protection on, because we have a lot of repos and we don't want to go through every single one and enable branch protection. `develop` in this particular repo was not protected, and refined-github was deleting every time I merged into master.

It really has very little to do with protected branches and way more to do with the extension doing something unexpected and requiring another click after a merge to get something back. I did not know this was a feature, so refined-github was actually requiring another click to restore this branch.

**Comment by busches** [comment]
> Many of the repos I'm on do not have branch protection on, because we have a lot of repos and we don't want to go through every single one and enable branch protection. `develop` in this particular repo was not protected, and refined-github was deleting every time I merged into master.

FYI, they did add an API to modify this so it can be automated for old repos and when creating new repos. This has made it super easy to protect a lot of repos, assuming you use standardized naming conventions between them.

> It really has very little to do with protected branches and way more to do with the extension doing something unexpected and requiring another click after a merge to get something back. I did not know this was a feature, so refined-github was actually requiring another click to restore this branch.

I'd disagree as that's the point of protected branches, you're still open to someone accidentally doing it if it's not protected. Either way, you should be able to get restore the branch with a click. The main issue was that you said you could not restore the branch sometime, which does seem like a GH bug and if you can reproduce it, should be reported to them.

It looks like you have multiple options here:
1) Protect long lived branches you don't want deleted. 
2) Click restore after it's deleted
3) Disable the feature
4) Remove the extension

I think option 1 is the correct course of action and it's what we did when we used long lived branches. We automated turning them on and then scripted the creation of newer feature branches, as we created them for a lot of repos each month.

Whatever you choose to do, thanks for taking the time to bring up the issue and I still think there needs to be further exploration into branches being deleted that cannot be restored, hopefully if it happens again you can report it to us or GH or both.

**Comment by qbunt** [comment]
Appreciate you taking the time to respond, I didn't know about the new API for automating this, thanks for the heads up.

I was able to restore without issue, it's more a UX decision of the extension that I question. I've run it for years without any kind of unexpected behavior, a feature was added at some point and I was surprised by it. For a browser extension to do something unexpected (esp. a deletion) is not great for building trust. I understand it's just a click, but so is deleting the branch.
